app/src/LightingManager.cpp

```diff
@@ -38,8 +38,8 @@ CHIP_ERROR LightingManager::Init()
     if (envGPIO == NULL || strlen(envGPIO) == 0)
     {
         ChipLogError(AppServer, "Environment variable not set or empty: %s", GPIO);
-        exit(-1);
-        // TODO: return an appropriate and fatal error
+
+        return CHIP_ERROR_INVALID_ARGUMENT;
     }
     ChipLogProgress(AppServer, "Using GPIO %s", envGPIO);
```

Comment:
It may be worth adding error handling to `stoi` as well, called few lines below.

Example exception handling for that: https://github.com/canonical/matter-pi-gpio-commander/blob/4df75fad2a661c2a93e73867909444b8cbce41bc/test-blink.cpp#L24-L33

Reply:
Added an issue: https://github.com/canonical/matter-pi-gpio-commander/issues/11